### PR TITLE
catppuccin: init at unstable-2023-10-09

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes

Consolidate catppuccin themes into a single package. Making it easier to pick and choose catppuccin themes and manage them in one package.

Related PRs: 
- https://github.com/NixOS/nixpkgs/pull/259511
- https://github.com/NixOS/nixpkgs/pull/259831
- https://github.com/NixOS/nixpkgs/pull/259665
- https://github.com/NixOS/nixpkgs/pull/259664
- https://github.com/NixOS/nixpkgs/pull/259661

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by khaneliman** [comment]
@Artturin this is what i got so far. what do you think?

File: pkgs/data/themes/catppuccin/default.nix [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    '' + lib.optionalString (lib.elem "bottom" themeList) ''
+      # bottom
+      mkdir -p $out/bottom
+      cp ${builtins.getAttr "bottom" sources}/"themes/${variant}.toml" $out/bottom/
+
+    '' + lib.optionalString (lib.elem "k9s" themeList) ''
+      # k9s
+      mkdir -p $out/k9s
+      cp ${builtins.getAttr "k9s" sources}/"dist/${variant}.yml" $out/k9s/
```
Comment (Artturin) [inline pkgs/data/themes/catppuccin/default.nix:97]:
```suggestion
      cp "${sources."k9s"}/dist/${variant}.yml" "$out/k9s/"
```

File: pkgs/data/themes/catppuccin/default.nix [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+, accent ? "blue"
+, variant ? "macchiato"
+, themeList ? [ "bat" "bottom" "btop" "k9s" "lazygit" ]
+}:
+let
+  pname = "catppuccin";
+
+  validAccents = [ "rosewater" "flamingo" "pink" "mauve" "red" "maroon" "peach" "yellow" "green" "teal" "sky" "sapphire" "blue" "lavender" ];
+  validVariants = [ "latte" "frappe" "macchiato" "mocha" ];
+  validThemes = [ "bat" "bottom" "btop" "k9s" "lazygit" ];
```
Comment (Artturin) [inline pkgs/data/themes/catppuccin/default.nix:13]:
```suggestion
let
  validThemes = [ "bat" "bottom" "btop" "k9s" "lazygit" ];
in
{ fetchFromGitHub
, lib
, stdenvNoCC
, accent ? "blue"
, variant ? "macchiato"
, themeList ? validThemes
}:
let
  pname = "catppuccin";

  validAccents = [ "rosewater" "flamingo" "pink" "mauve" "red" "maroon" "peach" "yellow" "green" "teal" "sky" "sapphire" "blue" "lavender" ];
  validVariants = [ "latte" "frappe" "macchiato" "mocha" ];
```

**Comment by khaneliman** [comment]
@Artturin looks like checks finished.

**Pull request merged**